<launch>
  <arg name="period" default="30" doc="period" />
  <arg name="world_frame_id" default="world" doc="world_frame_id" />
  <arg name="map_frame_id" default="map" doc="map_frame_id" />
  <arg name="map_topic" default="/osr_map" doc="the file name of the map to load." />
  <arg name="agent_info_topic" default="/osr_agent_info" doc="agent_info_topic." />
  <arg name="abnormal_topic" default="/osr_user_defined_abnormals" doc="abnormal_topic." />
  
  <arg name="illegal_parking_on" default="true" doc="illegal_parking_on" />
  <arg name="min_car_speed" default="1.0" doc="min_car_speed" />
  <arg name="illegal_parking_area" default="[36.12, 74.83, 14.73, 35.38, -19.79, 53.83, -42.67, 47.87]" doc="four points of one pair for illegal parking area (top_left_x, top_left_y, bottom_right_x, bottom_right_y)" />
  
  
  <arg name="high_thermal_on" default="true" doc="high_thermal_on" />
  <arg name="max_temperature" default="60" doc="max_temperature" />
  <arg name="high_thermal_area" default="[62.43, 37.82, 49.78, -14.33]" doc="four points of one pair for high thermal area (top_left_x, top_left_y, bottom_right_x, bottom_right_y)" />
  
  <arg name="high_elevation_on" default="true" doc="high_elevation_on" />
  <arg name="max_height" default="1.5" doc="max_height" />
  <arg name="high_elevation_area" default="[11.17, -19.22, -2.475, -22.37]" doc="four points of one pair for high elevation area (top_left_x, top_left_y, bottom_right_x, bottom_right_y)" />
  
  
  <arg name="viz_on" default="true" doc="viz_on" />
  
  
  <node name="user_defined_abnormal_node" pkg="osr_abnormals" type="user_defined_abnormal_node" output="screen">
      <param name="period" value="$(arg period)" />
      <param name="world_frame_id" value="$(arg world_frame_id)" />
      <param name="map_frame_id" value="$(arg map_frame_id)" />
      <param name="map_topic" value="$(arg map_topic)" />
      <param name="agent_info_topic" value="$(arg agent_info_topic)" />
      <param name="abnormal_topic" value="$(arg abnormal_topic)" />
      
      <param name="illegal_parking_on" value="$(arg illegal_parking_on)" />
      <param name="min_car_speed" value="$(arg min_car_speed)" />
      <rosparam param = "illegal_parking_area" subst_value="True">$(arg illegal_parking_area)</rosparam>
      
      <param name="high_thermal_on" value="$(arg high_thermal_on)" />
      <param name="max_temperature" value="$(arg max_temperature)" />
	  <rosparam param = "high_thermal_area" subst_value="True">$(arg high_thermal_area)</rosparam>
      
      <param name="high_elevation_on" value="$(arg high_elevation_on)" /> 
      <param name="max_height" value="$(arg max_height)" /> 
	  <rosparam param = "high_elevation_area" subst_value="True">$(arg high_elevation_area)</rosparam>
      
      <param name="viz_on" value="$(arg viz_on)" />
  </node>  

</launch>
